src/hotspot/share/utilities/globalDefinitions_xlc.hpp

```diff
@@ -38,6 +38,14 @@
 #include <stddef.h>
 #include <stdio.h>
 #include <stdlib.h>
+// Until IBM modifies its stdlib.h header we repair the critical lines ourself
+// replacing the #define malloc vec_malloc by the modified malloc Function declaration.
+#if (defined(__VEC__) || defined(__AIXVEC)) && defined(AIX) \
+    && defined(__open_xl_version__) && __open_xl_version__ >= 17
+  #undef malloc
+  extern void *malloc(size_t) asm("vec_malloc");
+#endif
```

Comment:
Wow!  And I don't mean that in a good way.  I'm not questioning whether this is correct, just commenting
on how crazy it seems that such a thing might be needed.

Reply:
The crazy thing is that `malloc` is defined! That means all places where we use the term malloc are getting replaced without such a workaround. (E.g. for log tags.)

Reply:
So, we do this only for malloc? Not for calloc, posix_memalign, realloc etc? What about free? 

As ugly as defining malloc is (and I remember QADRT), I hesitate about removing that define.

Removing that define and hard-coding it here assumes 1) our replacement is equivalent (ok, easy to check) 2) it will always be equivalent in future AIX versions 3) pointers it returns work with the unchanged free() and realloc() the system provides, and will always do so.

I don't know... I would not do this just to get rid of a warning.

Reply:
This one is not just to get rid of a warning. We get real test errors because malloc gets replaced by vec_malloc in log tags.

Reply:
> This one is not just to get rid of a warning. We get real test errors because malloc gets replaced by vec_malloc in log tags.

That does not invalidate my argument, nor does it answer my question. Those test errors could be also fixed by renaming the log tag. Maybe that would be the better way. 

Also, I'm curious, why does it not complain about "free", which is a logtag as well?

Reply:
I am basically worried that undefining malloc, even if it seems harmless now, exposes us to difficult-to-investigate problems down the road, since it depends on how the libc devs will reform those macros in the future. I would prefer a simple solution that does not depend on how the libc includes evolve.

Reply:
Is it possible to see the stdlib.h source code that is being a problem?  Maybe more eyes can come up
with a better solution, or at least come to a better understanding of why we have to go this way.

Reply:
Technically it's our fault.  The standard library is permitted to provide a
macro replacements for (nearly?) any function.  (C99 7.1.3/1 3rd bullet) But
it's really annoying.  Presumably AIX is only doing so for a subset of the
allocation functions, e.g. not "free" for example.

Having "malloc" defined as a macro seems like it should raise all kinds of
havoc, not just around log tags.  Consider our "os::malloc" function?  The
preprocessor knows nothing about C++ namespace qualifiers.